src_c/surface.c

```diff
@@ -1461,15 +1461,86 @@ surf_set_colorkey(pgSurfaceObject *self, PyObject *args)
         result = SDL_SetColorKey(surf, SDL_FALSE, color);
     }
     if (result == 0 && hascolor) {
-        result = SDL_SetSurfaceRLE(
+        printf("result == 0 && hascolor SDL_SetSurfaceRLE %i\n", flags & PGS_RLEACCEL);
+
+        // result = SDL_SetColorKey(surf, hascolor, color);
+        // SDL_UnlockSurface(surf);
+        int newresult = SDL_SetSurfaceRLE(
             surf, (flags & PGS_RLEACCEL) ? SDL_TRUE : SDL_FALSE);
+
+
+
+        // if (newresult == 0);
+        // {
+        //     printf("wwieuriwuerew, %i, %i\n", surf, newresult);
+        //     // return RAISE(pgExc_SDLError, SDL_GetError());
+        // }
+        // SDL_LockSurface(surf);
+        printf("--mustlock?, %i, SDL_RLEACCEL:%i, %i\n", SDL_MUSTLOCK(surf), surf->flags & SDL_RLEACCEL, surf->flags & PGS_RLEACCEL);
+
+        if (flags & PGS_RLEACCEL) {
+            result = SDL_SetColorKey(surf, hascolor | SDL_RLEACCEL, color);
+            // surf->flags |= SDL_RLEACCEL;
+        }
+        else {
+            result = SDL_SetColorKey(surf, hascolor, color);
+            // surf->flags &= ~SDL_RLEACCEL;
+        }
+
+        /*
+        Here we do something a bit hacky.
+         */
+        // SDL_Rect sdlrect3;
+        // sdlrect3.x = 0;
+        // sdlrect3.y = 0;
+        // sdlrect3.h = 0;
+        // sdlrect3.w = 0;
+
+        // // SDL_BlitSurface(surf, &sdlrect3, surf, &sdlrect3);
+        // SDL_LowerBlit(surf, &sdlrect3, surf, &sdlrect3);
+        // SDL_MapSurface(surf)
+
+        // SDL_UnlockSurface(surf);
+        printf("mustlock?, %i, SDL_RLEACCEL:%i, %i\n", SDL_MUSTLOCK(surf), surf->flags & SDL_RLEACCEL, surf->flags & PGS_RLEACCEL);
+
+        // SDL_RLESurface(surf);
+        // printf("new result? %i\n", surf->flags & SDL_RLEACCEL);
+        // printf("new result? %i, %i\n", result, surf->flags & SDL_RLEACCEL);
+        // result = SDL_SetColorKey(surf, hascolor, color);
+    } else {
+        result = SDL_SetColorKey(surf, hascolor, color);
     }
+
     if (result == 0) {
+        printf("before SDL_SetColorKey %i\n", surf->flags & SDL_RLEACCEL);
         result = SDL_SetColorKey(surf, hascolor, color);
     }
+    if (flags & PGS_RLEACCEL) {
+      /* we try and get the SDL_RLEACCEL flag to be set.
+      There's an SDL2 bug in SDL_SetSurfaceRLE, where it doesn't set this for some reason.
+      However, SDL_LowerBlit with a rect(0,0,0,0) does.
+      */
+      SDL_Rect sdlrect3;
+      sdlrect3.x = 0;
+      sdlrect3.y = 0;
+      sdlrect3.h = 0;
+      sdlrect3.w = 0;
+
+      SDL_Surface *surface;
+      surface = SDL_CreateRGBSurface(0, 1, 1, 32, surf->format->Rmask, surf->format->Gmask, surf->format->Bmask,
```

Comment:
Maybe we can create a global surface to use for this hack. So it can be reused more easily, and so it's a bit faster.